### PR TITLE
Backport workflow changes to 0.28.x branch

### PR DESCRIPTION
**Pull request by kevinbackhouse**
These are changes that were made on the main branch but haven't been backported to 0.28.x yet. The only difference now is that I didn't backport [`special_noFilesystemAccess`](https://github.com/Exiv2/exiv2/blob/a08de6fa863c82cbc3ea86226d5ef022dbbbe871/.github/workflows/on_PR_linux_special_builds.yml#L130) because I think that's a new feature on the main branch.

### CODE REVIEW

File: .github/workflows/nightly_Linux_distributions.yml
Comment (kmilos) [inline .github/workflows/nightly_Linux_distributions.yml:1]: I think we could remove the nightly and release workflows from 0.28.x completely?

Comment (kevinbackhouse) [reply .github/workflows/nightly_Linux_distributions.yml:1]: I think the release workflow is useful for building the artifacts when we do a new release. You're right that the nightly is unnecessary, so I'll delete it.

Comment (kmilos) [reply .github/workflows/nightly_Linux_distributions.yml:1]: Btw, the release one is currently manually disabled / non-functional?

Comment (kevinbackhouse) [reply .github/workflows/nightly_Linux_distributions.yml:1]: Oh, you're right. I wonder what happened there. It'll be convenient to get it working before I create the new release.

File: .github/workflows/codeql-analysis.yml
Comment (kmilos) [inline .github/workflows/codeql-analysis.yml:1]: We should probably change this one to `branches: [main, 0.28.x]` (or juts main?), and fix this on main as well...

File: .github/workflows/on_push_ExtraJobsForMain.yml
Comment (kmilos) [inline .github/workflows/on_push_ExtraJobsForMain.yml:1]: Remove as well, only runs on main

Comment (kmilos) [reply .github/workflows/on_push_ExtraJobsForMain.yml:1]: Also remove codesee-arch-diagram.yml

File: .github/workflows/on_PR_windows_matrix.yml
Comment (kmilos) [inline .github/workflows/on_PR_windows_matrix.yml:1]: This one also runs on main only for now, so either we remove it, or add 0.28.x branch as well?

File: .github/workflows/nightly_Linux_distributions.yml
Comment (kevinbackhouse) [reply .github/workflows/nightly_Linux_distributions.yml:1]: It's failing while building the samples. Based on some local testing, I think it'll pass if I change the setting to `-DEXIV2_ENABLE_WEBREADY=ON`. Alternatively, I could add `-DEXIV2_BUILD_SAMPLES=OFF`. I'm not sure which is better.

Comment (kmilos) [reply .github/workflows/nightly_Linux_distributions.yml:1]: I guess disable the samples? Don't think those are supposed to be supported/production quality code...

**Comment by kmilos** [comment]
Shall we do one last compare after the last few commits to main?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ jobs:
 
       - name: Build packaged release
         run: |
-          cmake --preset win-release -S . -B build -DEXIV2_TEAM_PACKAGING=ON -DEXIV2_BUILD_DOC=ON -DEXIV2_ENABLE_WEBREADY=OFF -DEXIV2_ENABLE_CURL=OFF -DEXIV2_BUILD_SAMPLES=OFF
+          cmake --preset win-release -S . -B build -DEXIV2_TEAM_PACKAGING=ON -DEXIV2_BUILD_DOC=ON -DEXIV2_ENABLE_WEBREADY=OFF -DEXIV2_ENABLE_CURL=OFF
```
Comment (kmilos) [inline .github/workflows/release.yml:132]: How come this went back?!

Comment (kevinbackhouse) [reply .github/workflows/release.yml:132]: Looks like I made a mistake on the main branch in #2921. There's two instances that need to be updated but I only did one.

**Review by kmilos — APPROVED** [review]

**Pull request merged**